File: CS/WindowsFormsApplication18/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Base;
using MRUComboBox;

namespace WindowsFormsApplication18
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}

		private void barEditItem1_EditValueChanged(object sender, EventArgs e)
		{
			if ( repositoryItemMRUComboBox1.Items.Contains(barEditItem1.EditValue) )
				repositoryItemMRUComboBox1.StoreRecentItem(barEditItem1.EditValue);
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			this.carsTableAdapter.Fill(this.carsDBDataSet.Cars);
		}

		private void gridView1_CellValueChanged(object sender, CellValueChangedEventArgs e)
		{
			if ( e.Column.ColumnEdit is RepositoryItemMRUComboBox )
			{
				RepositoryItemMRUComboBox columnEdit = (RepositoryItemMRUComboBox)e.Column.ColumnEdit;
				if ( columnEdit.Items.Contains(e.Value) )
					columnEdit.StoreRecentItem(e.Value);
			}
		}
	}
}

File: VB/WindowsFormsApplication18/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Base
Imports MRUComboBox

Namespace WindowsFormsApplication18
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub barEditItem1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles barEditItem1.EditValueChanged
			If repositoryItemMRUComboBox1.Items.Contains(barEditItem1.EditValue) Then
				repositoryItemMRUComboBox1.StoreRecentItem(barEditItem1.EditValue)
			End If
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            'TODO: This line of code loads data into the 'CarsDBDataSet.Cars' table. You can move, or remove it, as needed.
            Me.CarsTableAdapter.Fill(Me.CarsDBDataSet.Cars)
        End Sub

		Private Sub gridView1_CellValueChanged(ByVal sender As Object, ByVal e As CellValueChangedEventArgs) Handles gridView1.CellValueChanged
			If TypeOf e.Column.ColumnEdit Is RepositoryItemMRUComboBox Then
				Dim columnEdit As RepositoryItemMRUComboBox = CType(e.Column.ColumnEdit, RepositoryItemMRUComboBox)
				If columnEdit.Items.Contains(e.Value) Then
					columnEdit.StoreRecentItem(e.Value)
				End If
			End If
		End Sub
	End Class
End Namespace